src/interpreter/mod.rs

```diff
@@ -53,10 +59,21 @@ impl Into<String> for Error {
 			Error::Env(s) => s,
 			Error::Native(s) => s,
 			Error::Trap(s) => format!("trap: {}", s),
+			Error::User(e) => format!("user: {}", e),
 		}
 	}
 }
 
+/// Dummy user error.
+#[derive(Debug, Clone, PartialEq)]
+pub struct DummyCustomUserError;
```

Comment:
So you're suggesting to add additional namespace (error) and move this struct over there (maybe I got it wrong)? Looks like additional headache for me, but if you feel this is ok - please insist.
For now I have renamed it to `DummyUserError` && added alias `DummyError` for `Error<DummyUserError>`. Unfortunately, because of this issue: https://github.com/rust-lang/rust/issues/26264 you still need to import both `Error` && `DummyError`, if you want to access `Error` variants